tket/src/Architecture/Architecture.hpp

```diff
@@ -67,48 +67,68 @@ class ArchitectureBase : public T {
   using T::node_exists;
   using T::nodes;
   using T::T;
+};
+
+class Architecture : public ArchitectureBase<graphs::DirectedGraph<Node>> {
+ public:
+  /** Construct an empty Architecture. */
+  Architecture() : ArchitectureBase() {}
+
+  /** Construct an Architecture with given nodes and no edges. */
+  explicit Architecture(const std::vector<Node> &nodes)
+      : ArchitectureBase(nodes) {}
+
+  /** Construct an Architecture with given edges. */
+  explicit Architecture(const std::vector<std::pair<Node, Node>> &edges)
+      : ArchitectureBase(edges) {}
 
   /**
    * Construct from a vector of pairs of indices in the default register.
    */
-  explicit ArchitectureBase(
-      const std::vector<std::pair<unsigned, unsigned>> &edges)
-      : ArchitectureBase(as_nodepairs(edges)) {}
+  explicit Architecture(const std::vector<std::pair<unsigned, unsigned>> &edges)
+      : Architecture(as_nodepairs(edges)) {}
 
-  /** compute articulation points of Architecture
-   *  if `subarc` is given, the articulation that do not affect the
-   * sub-architecture are ignored */
+  /**
+   * Nodes that cannot be removed without breaking connectivity.
+   */
   node_set_t get_articulation_points() const;
-  node_set_t get_articulation_points(const ArchitectureBase &subarc) const;
 
-  /* returns new Architecture that is generated by a subset of nodes of `this`
+  /**
+   * Nodes that cannot be removed without breaking connectivity of the subarc.
+   */
+  node_set_t get_articulation_points(const Architecture &subarc) const;
```

Comment:
Could `subarc` be of type `ArchitectureBase`?

Reply:
The implementation actually uses the BGL methods so I think not easily.